sirepo/package_data/static/js/sirepo-components.js

```diff
@@ -3137,24 +3137,73 @@ SIREPO.app.directive('sbatchLoginModal', function() {
 
 });
 
-SIREPO.app.directive('sbatchCoresAndHours', function(appState) {
+SIREPO.app.directive('sbatchOptions', function(appState) {
     return {
         restrict: 'A',
         scope: {
-            simState: '=sbatchCoresAndHours',
+            simState: '=sbatchOptions',
         },
         template: [
             '<div class="clearfix"></div>',
             '<div style="margin-top: 10px" data-ng-show="showCoresAndHours()">',
                 '<div data-model-field="\'sbatchHours\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"></div>',
                 '<div data-model-field="\'sbatchCores\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"></div>',
+                '<div data-model-field="\'sbatchQueue\'" data-model-name="simState.model" data-label-size="3" data-field-size="3"  data-ng-click="sbatchQueueFieldIsDirty = true"></div>',
             '</div>',
         ].join(''),
-        controller: function($scope) {
+        controller: function($scope, $element) {
+            $scope.sbatchQueueFieldIsDirty = false;
+            var model = appState.models[$scope.simState.model];
+            var coresPerNode = 32;
+            var max = {
```

Comment:
This is nersc, not general. For a departmental cluster, sbatch may have only one queue, for example. I think we are going to want to send this configuration from the server. 

We are hacking this into auth_state now so we might as well pile it in there. It's an abstraction (job_run_mode_map). This could contain all the configuration. We will have to get the data from sbatch.cfg, which is in the supervisor. Therefore, we may have to move it into job.py. 

Thoughts?

Reply:
https://github.com/radiasoft/sirepo/pull/2511/commits/579c3fbbfc59f452081b73a6b75a90de77873122

The queue values are hardcoded. Should they come from pkconfig?

Reply:
That's fine. They aren't changing. As we add clusters, we'll figure out the model for configuration.